src/PrestaShopBundle/Form/Admin/Improve/International/Localization/LocalizationConfigurationType.php

```diff
@@ -0,0 +1,184 @@
+     */
+    private $countryDataProvider;
+
+    /**
+     * @var CurrencyDataProvider
+     */
+    private $currencyDataProvider;
+
+    /**
+     * @var TimezoneRepository
+     */
+    private $timezoneRepository;
+
+    /**
+     * @param LegacyContext $legacyContext
+     * @param LanguageDataProvider $languageDataProvider
+     * @param CountryDataProvider $countryDataProvider
+     * @param CurrencyDataProvider $currencyDataProvider
+     * @param TimezoneRepository $timezoneRepository
+     */
+    public function __construct(
+        LegacyContext $legacyContext,
+        LanguageDataProvider $languageDataProvider,
+        CountryDataProvider $countryDataProvider,
+        CurrencyDataProvider $currencyDataProvider,
+        TimezoneRepository $timezoneRepository
+    ) {
+        $this->languageDataProvider = $languageDataProvider;
+        $this->countryDataProvider = $countryDataProvider;
+        $this->legacyContext = $legacyContext;
+        $this->currencyDataProvider = $currencyDataProvider;
+        $this->timezoneRepository = $timezoneRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('default_language', ChoiceType::class, [
+                'choices' => $this->getLanguageChoices(),
+            ])
+            ->add('detect_language_from_browser', SwitchType::class)
+            ->add('default_country', ChoiceType::class, [
+                'choices' => $this->getCountryChoices(),
+            ])
+            ->add('detect_country_from_browser', SwitchType::class)
+            ->add('default_currency', ChoiceType::class, [
+                'choices' => $this->getCurrencyChoices(),
+            ])
+            ->add('timezone', ChoiceType::class, [
+                'choices' => $this->getTimezoneChoices(),
+            ])
+        ;
+    }
+
+    /**
+     * Get available language choices
+     *
+     * @return array
+     */
+    private function getLanguageChoices()
+    {
+        $languages = $this->languageDataProvider->getLanguages();
+        $choices = [];
+
+        foreach ($languages as $language) {
+            $choices[$language['name']] = $language['id_lang'];
+        }
+
+        return $choices;
+    }
+
+    /**
+     * Get available country choices
+     */
+    private function getCountryChoices()
+    {
+        $contextLanguage = $this->legacyContext->getLanguage();
+        $countries = $this->countryDataProvider->getCountries($contextLanguage->id);
+        $choices = [];
+
+        foreach ($countries as $country) {
+            $choices[$country['name']] = $country['id_country'];
+        }
+
+        return $choices;
+    }
+
+    /**
+     * Get available currency choices
+     *
+     * @return array
+     */
+    private function getCurrencyChoices()
+    {
+        $currencies = $this->currencyDataProvider->getCurrencies(
+            $asObjects = false,
```

Comment:
same here, you don't need the variables... I guess your intent is to describe the function :)

Once these function will be moved to the providers, you won't need it anymore :)